### DISCUSSION TITLE
Contract verification failed (FundMe - SEPOLIA)

### DISCUSSION THREAD

**Discussion by worca333** [Cyfrin/foundry-full-course-cu#1945] [Q&A]
I tried to manually verify `FundMe` contract after deploying it in sepolia network.
But it failed due to 3rd party interface file (chainlink) and I am not sure how to include that. Note that foundry made a path alias for these chainlink contracts. Full error:

```
ParserError: Source "@chainlink/contracts/src/v0.8/shared/interfaces/AggregatorV3Interface.sol" not found: File import callback not supported
 --> FundMe.sol:6:1:
  |
6 | import {AggregatorV3Interface} from "@chainlink/contracts/src/v0.8/shared/interfaces/AggregatorV3Interface.sol"
  | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

ParserError: Source "@chainlink/contracts/src/v0.8/shared/interfaces/AggregatorV3Interface.sol" not found: File import callback not supported
 --> PriceConverter.sol:4:1:
  |
4 | import {AggregatorV3Interface} from "@chainlink/contracts/src/v0.8/shared/interfaces/AggregatorV3Interface.sol"
  | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
```

Would someone please help?

**Comment by EngrPips** [comment]
Hello @worca333, Have you traced the path in your working directory and confirmed that the file is actually on that path?

This is the path where it is for my own project. 

```
import {AggregatorV3Interface} from "@chainlink/contracts/src/v0.8/interfaces/AggregatorV3Interface.sol";
```

**Reply by worca333** [reply]
Sure, though it resides in a slightly different path than it looks.
Foundry did some remapping using its config like 
```
remappings = [
    '@chainlink/contracts/=lib/chainlink-brownie-contracts/contracts/',
]
```

After all, the verification process needs to verify this interface as well.

**Reply by EngrPips** [reply]
This is my path to the file shown in the image below. Show yours, too.

<img width="1440" alt="Screenshot 2024-07-04 at 17 33 35" src="https://github.com/Cyfrin/foundry-full-course-cu/assets/95760918/560c9780-8fc1-4c63-811e-a67182250ed9">

**Reply by worca333** [reply]
Sure.
![image](https://github.com/Cyfrin/foundry-full-course-cu/assets/167474153/92478717-7380-4d58-a6aa-316690fdfcb6)

**Reply by EngrPips** [reply]
You didn't look closely, obviously, There is no `shared` in the path you showed me, So please use this path `"@chainlink/contracts/src/v0.8/interfaces/AggregatorV3Interface.sol"` for your import, and I think you will be good

**Reply by worca333** [reply]
Good catch, man! But I guess it's not essential.
Being irony, we have the same interface under shared folder as well, that's why compilation was successful.

![image](https://github.com/Cyfrin/foundry-full-course-cu/assets/167474153/6f3d2c1b-59ab-4153-a14d-fe24e83e9f79)

**Reply by EngrPips** [reply]
If the `shared` works for you, we are good.

**Reply by worca333** [reply]
@EngrPips 
Maybe I wasn't clear enough about the issue.
I successfully deployed the contract in the Sepolia network and here's the address: https://sepolia.etherscan.io/address/0xd94d92809782284B95FF847B78355dA3eE378020#code
While I was trying to verify it to reveal ABI, I got the error.

**Reply by EngrPips** [reply]
I got the issue @worca333, and I am trying to resolve it and get back to you. I tried to verify this myself, and I ran into the same issue so we are in this together now, lol. I will surely come back here with my solution.

**Reply by EngrPips** [reply]
Hey @worca333, So I verified my `FundMe` contract. you can see it here `https://sepolia.etherscan.io/address/0xD7E259ba80F0F1dDF5422629632F5f6A9BC59E8F#code`
I did through the terminal though if you are fine with that.

**Reply by worca333** [reply]
@EngrPips 
Via terminal means using `forge ... --verify`?
Otherwise, I'd love to get the details.

**Reply by EngrPips** [reply]
Yeah, using `--verify` and passing the `etherscan-api-key` would try that, and let's see if it works for you.

**Reply by worca333** [reply]
@EngrPips 
Of course, that I've tried and it worked.
Technically, we would not verify the contracts of our own.
Just this happens when we want to verify some after they go live.

**Reply by EngrPips** [reply]
I am sure there would be a way to make the verification work but honestly, I can't dedicate time to digging and researching that because of other important stuff I need to do. But it it happens we deploy a real codebase to production and we didn't verify on deployment definitely we would find the walkaround to make sure it is verified.
